lightning/backend/backend.go

```diff
@@ -388,3 +397,16 @@ type Rows interface {
 	// capacity with the current instance. The typical usage is `x = x.Clear()`.
 	Clear() Rows
 }
+
+type EngineWriter interface {
+	AppendRows(
+		ctx context.Context,
+		tableName string,
+		columnNames []string,
+		commitTS uint64,
+		rows Rows,
+	) error
+	Close() error
+	AddProducer()
+	Done()
```

Comment:
i think the `EngineWriter` interface doesn't need `AddProducer` and `Done` methods.
the `produceWg WaitGroup` could just be a member of `LocalEngineWriter`.

Reply:
Ok. I will refactor it.

Reply:
Done